### PR TITLE
OAuth

### PR DESCRIPTION
**Pull request by voidxnull**
Adds a new component with OAuth buttons, new redux actions, and a few utility functions.

Related: https://github.com/FreeFeed/freefeed-server/pull/296

### Configuration
The only config option is:
```
config.auth.oauth: ['facebook', 'google', 'github']
```
It can be set to an array of enabled providers or `false` to disable OAuth entirely.

┆Issue is synchronized with this [Trello card](https://trello.com/c/tZg24SZj)

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -9,7 +9,8 @@ const config = {
     cookieDomain: 'localhost',
     tokenPrefix: 'freefeed_',
     userStorageKey: 'USER_KEY',
-    sentinel: null // keep always last
+    sentinel: null, // keep always last
+    oauth: false,
```
Comment (dsumin) [inline src/config.js]: This should probably go before 'sentinel: null, // keep always last'

Comment (voidxnull) [reply src/config.js]: Done.

**Comment by voidxnull** [comment]
Done. It might need some visual touch ups though. A few screenshots:
freefeed_oauth_btns
freefeed_oauth_accs
freefeed_fb_friends_1
freefeed_fb_friends_2

**Comment by voidxnull** [comment]
New external accounts editor:
freefeed_new_oauth_accounts

**Pull request closed without merging**